### ISSUE TITLE
Optimization: Expose `ucp_context_attr.request_size` as a constant

### ISSUE THREAD

**Issue by raphaelcohn** [openucx/ucx#2392]
`ucp_context_attr.request_size` is used to allocate stack space for a request before calling a method such as `ucp_tag_send_nbr`. However, this reservation is intended to use `alloca`, which can't be optimized as efficiently by LLVM as a known-in-advance fixed-size reservation (eg of a fixed size array).

Additionally, wrapping languages (such as Rust) do not support `alloca`.

Given that this field value never changes - it is a compilation time constant - is there a way this value be exposed in the API? (eg as a `#define`)? Obviously, this isn't simple, but it would make for more optimizations - and that is a good thing for users of UCX. Alternatively, we could expose an over-estimate and use a test case to make sure it's always sufficient.

(As an aside, in my Rust wrapper, I'm going to just hard code this value to 256 [or another power of two] - the current 1.3 branch on Musl / Linux for my choice is less than this).

**Comment by yosefe** [comment]
exposing this field is prone to breaking ABI compatibility, if such compatibility is not required - a user can detect the size of the request by a ./configure script (and hope it would not grow)

**Comment by raphaelcohn** [comment]
Yep, it will break between versions of UCX; I'd argue that was a good thing. If I compile for a particular version of UCX (or any other library), then I *should not* expect my code to work with any other version with the *exception* of minor security fixes.

In C land, this view of mine is best enforced by using static linkage. It's far more secure and far more robust in the face of complex administrative challenge.

What would be involved in adding a little something to a ./configure that could work when UCX is being cross-compiled (another view point: all libraries should always be treated as if cross-compiled. It stops build system assumptions making there way into production).
